linera-execution/src/wasm/wasmer.rs

```diff
@@ -253,17 +257,21 @@ impl CachedContractModule {
 
     /// Creates a new [`Engine`] to compile a contract bytecode.
     fn create_compilation_engine() -> Engine {
-        let mut compiler_config = Singlepass::default();
-        compiler_config.canonicalize_nans(true);
+        #[cfg(not(web))]
+        {
+            let mut compiler_config = wasmer::Singlepass::default();
+            compiler_config.canonicalize_nans(true);
+
```

Comment:
nit: remove newline?

Reply:
I think the newline is important here to visually separate the setup from the use of `compiler_config`.

(Ideally it would be a builder! And then the whole thing would be one expression.)

Reply:
IMHO this is all doing one thing

Reply:
It's small enough that I'm not particularly bothered by it.  I think if we decide to make changes elsewhere I'll include it in the next push — otherwise I'd like to merge it without waiting another hour for CI so I can point `linera-web` to it :sweat_smile: